os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c

```diff
@@ -1350,7 +1350,7 @@ extern void __libc_init_array(void);
 #ifdef CONFIG_PLATFORM_TIZENRT_OS
 #ifdef CONFIG_ARMV8M_MPU
 	/* Initialize number of mpu regions for board specific purpose */
-	mpu_set_nregion_board_specific(1);
+	mpu_set_nregion_board_specific(mpu_entry_alloc());
```

Comment:
does mpu_entry_alloc() reserve one more region here?
because for region 0 it was mentioned entry_alloc reserve the region

Reply:
Every time we call entry_alloc(), it will reserve one region internally. 
For example if rtk is using below regions, 
xxx = entry_alloc() ==> region 0
psram = entry_alloc() ==> region 1

So, when we call entry_alloc() again, it returns value 2, which is correct for our mpu_set_nregion_board_specific() API.
After this point, realtek code is not allocating more regions.

Reply:
yes correct, to know the already allocated number we using secure allocation call.  The function name and the purpose we use seems mismatching. 
Can we get mpu_entry from app_mpu_nocache_init and pass here?

Reply:
Ok. I modified it to use a global variable.